src/Components/Weather/WeatherListItem.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from 'react-eva-icons';
+import constant from 'data/const';
+
+class WeatherListItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      city: undefined,
+      temp: undefined,
+    };
+  }
+
+  componentDidMount() {
+    const location = this.props.location;
+    const URL = `http://api.openweathermap.org/data/2.5/weather?q=${location}&appid=${constant.WEATHER_API_KEY}&units=metric`;
+    fetch(URL).then(res => res.json()).then(json => {
+      this.setState({
+        city: json.name,
+        temp: json.main.temp,
+      });
+    });
+  }
+
+  render() {
+    const isDataLoaded = this.state.city && this.state.temp;
+    if (!isDataLoaded) return <div>Loading</div>;
+    return (
+      <div>
+        {this.state.city} - {this.state.temp}°
+      </div>
+    );
+  }
+}
```

Comment:
I don't like this solution, I've tried to write condition in `return()` but all project crashed after this

Reply:
what about 

return isDataLoaded ? <div>{this.state.city} - {this.state.temp}°</div> : <div>Loading</div>;

Reply:
doesn't work. can you help me with this tomorrow?

Reply:
```
render() {
    return (
      <div>
        {this.state.city && this.state.temp ? `${this.state.city} - ${this.state.temp}` : 'Loading'}
      </div>
    );
  }
```